epix/io.py

```diff
@@ -80,75 +81,106 @@ def loader(directory,
+                    f' & ((zp >= {outer_cylinder["min_z"] * 10}) & (zp < {outer_cylinder["max_z"] * 10}))')
     else:
         cut_string = None
 
-    # Read in data, convert mm to cm and perform a first cut if specified:
-    interactions = ttree.arrays(column_names,
-                                cut_string,
-                                aliases=alias,
-                                **kwargs_uproot_arrays)
-    eventids = ttree.arrays('eventid', **kwargs_uproot_arrays)
-    eventids = ak.broadcast_arrays(eventids['eventid'], interactions['x'])[0]
-    interactions['evtid'] = eventids
+
+    if file.endswith(".root"):
+        ttree, n_simulated_events = _get_ttree(directory, file_name)
+
+        if arg_debug:
+            print(f'Total entries in input file = {ttree.num_entries}')
+            cutby_string='output file entry'
+            if cut_by_eventid:
+                cutby_string='g4 eventid'
+
+            if kwargs['entry_start'] is not None:
+                print(f'Starting to read from {cutby_string} {kwargs["entry_start"]}')
+            if kwargs['entry_stop'] is not None:
+                print(f'Ending read in at {cutby_string} {kwargs["entry_stop"]}')
+
+        # If user specified entry start/stop we have to update number of
+        # events for source rate computation:
+        if kwargs['entry_start'] is not None:
+            start = kwargs['entry_start']
+        else:
+            start = 0
+
+        if kwargs['entry_stop'] is not None:
+            stop = kwargs['entry_stop']
+        else:
+            stop = n_simulated_events
+        n_simulated_events = stop - start
+
+        if cut_by_eventid:
+            # Start/stop refers to eventid so drop start drop from kwargs
+            # dict if specified, otherwise we cut again on rows.
+            kwargs.pop('entry_start', None)
+            kwargs.pop('entry_stop', None)
+
+        # Columns to be read from the root_file:
+        column_names = ["x", "y", "z", "t", "ed",
+                        "type", "trackid",
+                        "parenttype", "parentid",
+                        "creaproc", "edproc"]
+
+        # Conversions and parameters to be computed:
+        alias = {'x': 'xp/10',  # converting "geant4" mm to "straxen" cm
+                'y': 'yp/10',
+                'z': 'zp/10',
+                'r': 'sqrt(x**2 + y**2)',
+                't': 'time*10**9'
+                }
+
+        # Read in data, convert mm to cm and perform a first cut if specified:
+        interactions = ttree.arrays(column_names,
+                                    cut_string,
+                                    aliases=alias,
+                                    **kwargs)
+        eventids = ttree.arrays('eventid', **kwargs)
+        eventids = ak.broadcast_arrays(eventids['eventid'], interactions['x'])[0]
+        interactions['evtid'] = eventids
+
+    elif file.endswith(".csv"):
```

Comment:
Same here. Thats also why I would suggest to make things a class. Since some stuff will be shared by the two and hence object oriented programing might help to make it a bit cleaner.